### PR TITLE
[RW-6866][risk=no] Enforce a 5K sample limit on genomic extraction

### PR DESCRIPTION
**Pull request by calbach**
* Display the error message returned in JSON from the server, if any - currently these are intended for client consumption, so should be comprehensible by the end user
* Currently we only expect this to happen in two cases:
  * 412: no WGS samples
  * 412: too many WGS samples
* If a client error is encountered, gray out the "extract" button - this is not retryable

Other changes:
* Make error and warning mutually exclusive, showing both looked confusing
* Update language to reference the "genomic job history" panel, rather than the "workspace storage" panel 
* Make styling consistent between warning and error callout

Screenshots:

Error (induced by setting a limit of 1 sample) via dataset card modal:
image

Error (induced by setting a limit of 1 sample) via dataset builder:
image

### CODE REVIEW

File: ui/src/app/pages/data/data-set/genomic-extraction-modal.spec.tsx [hunk 1]
```diff
@@ -124,7 +125,7 @@ describe('GenomicExtractionModal', () => {
     expect(warning.text()).toContain("Last time a VCF extract was attempted for this workflow, it failed.");
   });
 
-  it('should not show a warning message with no succeded, failed, or running extracts for this dataste', () => {
+  it('should not show a warning message with no succeeded, failed, or running extracts for this dataset', () => {
```
Comment (als364) [inline ui/src/app/pages/data/data-set/genomic-extraction-modal.spec.tsx:128]: uh oh these were my typos lol. thanks for fixing

**Review by als364 — APPROVED** [review]
lgtm. How did the 5k limit get decided?

**Comment by calbach** [comment]
Oops sorry - missed your comment:

5k - somewhat arbitrary, but hopefully low enough that it should not cause issues. With more testing we may be able to increase this. Basically I wanted to pick a value here where someone is not wasting money trying to re-extract a significant fraction of the samples, where they could have instead just pointed at the pre-materialized merged VCF. ~30% felt like roughly the sweet spot.

But I think we do need some more data collection here, since I don't know whether 5K will even succeed.

**Comment by als364** [comment]
That makes sense, thank you!

**Pull request merged**